### PR TITLE
School library: Setup associations

### PR DESCRIPTION
**Pull request by ernestmusong**
## In this PR, I have implemented the following:

- Added classes **Classroom**, **Book**, and **Rental** and their associations with each other.

### CODE REVIEW

**Review by Ismailco — COMMENTED** [review]
Hi @ernestmusong, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that uses the Microverse README template however, kindly make sure to include all the required headers in your README, and fix the LICENSE link for now it redirects to a 404 page.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by ernestmusong** [comment]
> Hi @ernestmusong,
> 
> While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
> 
> ## Invalid Code Review Request
> You have submitted a project, that uses the Microverse README template however, kindly make sure to include all the required headers in your README, and fix the LICENSE link for now it redirects to a 404 page.
> 
> Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
> 
> Cheers, and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _Invalid Code Review Request **does not count** into the code reviews limit._



> Hi @ernestmusong,
> 
> While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
> 
> ## Invalid Code Review Request
> You have submitted a project, that uses the Microverse README template however, kindly make sure to include all the required headers in your README, and fix the LICENSE link for now it redirects to a 404 page.
> 
> Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
> 
> Cheers, and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _Invalid Code Review Request **does not count** into the code reviews limit._

Thank you @Ismailco for letting me know. I have now fixed the LICENSE link in the readme. However, some headers in the readme such as **Live demo** and **Live presentation** are omitted because the project is not yet deployed.

**Comment by Ismailco** [comment]
@ernestmusong, the only header that is not required for this milestone in `Live demo` and `Live presentation`, actually you should remove these two headers

File: classes/student.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
+require_relative 'nameable'
 class Student < Person
   attr_reader :classroom
```
Comment (Olamarx) [inline classes/student.rb:5]: - [ ] I can see you set the getter, good job. However, you forgot to add a method that serves as the 'belongs-to' side (a student belongs to a classroom). Kindly work on this, let me know if you don't understand so I can explain further.

File: classes/person.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+require 'securerandom'
+require_relative 'nameable'
+require_relative '../decorators/capitalize_decorator'
+require_relative '../decorators/trimer_decorator'
+
+class Person < Nameable
+  attr_accessor :name, :age
+  attr_reader :id, :rentals
```
Comment (Olamarx) [inline classes/person.rb]: - [ ] :rentals should be for Getters and Setters so I will advise you to use the shortcut of attr_accessor.

File: classes/person.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  def initialize(name: 'Unknown', age: 0, parent_permission: true)
+    super()
+    @id = SecureRandom.uuid
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    @rentals = []
```
Comment (Olamarx) [inline classes/person.rb:17]: - [ ] Good one initializing the @rentals. but a method for adding rentals should be present also, and in the method, the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association. You can use `Rental.new(book, date, self)`.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @ernestmusong,

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- good use of association ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ernestmusong** [comment]
Thank you @Olamarx for your feedback. I have now implemented the above changes required.

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @ernestmusong 

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀

image


## Highlights
- Good use of SQL queries✔️ 
- Kudus to heeding the required and optional changes of the last reviewer 👍 

## Optional Suggestions 💡
While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really enhance your application.
Cheers, and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread, and don't forget to tag me so I'll receive the notification.

As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**